README.md

```diff
@@ -72,6 +72,58 @@ Instructions for the experimental Carthage distribution are at
 Instructions for installing binary frameworks via
 [Rome](https://github.com/CocoaPods/Rome) are at [Rome](Rome.md).
 
+### Using Firebase from a Framework or a library
+
+Currently all source code and binary Firebase SDKs are compiled as **static** frameworks.
+It may be not too important detail if Firebase SDKs are used from an app target directly,
+but must be taken into account if Firebase is used from another library or a framework.
+The key difference is in the way static and dynamic linking works. Your framework itself may
+be either static or dynamic. Let's consider these two options in more details.
+
+#### Using Firebase SDKs in from dynamic framework
+
+When a dynamic framework/library is compiled each static dependency symbols
+are added to the dynamic framework binary. It is not really important unless you need to
+use both the dynamic framework and one of its static dependencies directly in you app. In this case if you attempt to link
+the same static framework/library to both the app and the dynamic framework you will end up with the static framework symbols added to both the app and the dynamic framework binaries. It means that when you app launches
+it will have 2 copy of the static framework symbols. This leads to undefined behavior (especially when
+different versions of the static framework are linking to the app and the dynamic framework).
```

Comment:
For example, a `dispatch_once` may or may not do the right initialization since there are now two entities to initialize.